iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/helpers/Tools.java

```diff
@@ -124,6 +124,112 @@ public static boolean isCause(Class<? extends Throwable> possibleExceptionCause,
         return possibleExceptionCause.isInstance(exceptionToSearch) || (exceptionToSearch != null && isCause(possibleExceptionCause, exceptionToSearch.getCause()));
     }
 
+    private static RegistryManager registryManager;
+
+    public static RegistryManager getRegistyManager(String iotHubConnectionString) throws IOException
+    {
+        if (registryManager == null)
+        {
+            RegistryManagerOptions options = RegistryManagerOptions.builder().httpReadTimeout(HTTP_READ_TIMEOUT).build();
+            registryManager = RegistryManager.createFromConnectionString(iotHubConnectionString, options);
+        }
+
+        return registryManager;
+    }
+
+    public static TestDeviceIdentity getTestDevice(String iotHubConnectionString, IotHubClientProtocol protocol, AuthenticationType authenticationType) throws URISyntaxException, IOException, IotHubException, GeneralSecurityException, InterruptedException
```

Comment:
Both ```getTestDevice``` and ```getTestModule``` existed in several places within our e2e tests, so the big change for this PR is to centralize this logic to one place. That way we can make edits (test identity recycling, bulk creation) here and have all tests get those edits.

For now though, this does not change how many or how me make the test identities